File: ios/Classes/HttpProxyPlugin.m
#import "HttpProxyPlugin.h"

@implementation HttpProxyPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"com.lm.http.proxy"
            binaryMessenger:[registrar messenger]];
  HttpProxyPlugin* instance = [[HttpProxyPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getProxyHost" isEqualToString:call.method]) {
      CFDictionaryRef proxySettings = CFNetworkCopySystemProxySettings();
      NSDictionary *dictProxy = (__bridge_transfer id)proxySettings;
      //是否开启了http代理
      if ([[dictProxy objectForKey:@"HTTPEnable"] boolValue]) {
          NSString *proxyHost = [dictProxy objectForKey:@"HTTPProxy"];
          result(proxyHost);
      }else{
        result(nil);
      }
    } else if ([@"getProxyPort" isEqualToString:call.method]) {
      CFDictionaryRef proxySettings = CFNetworkCopySystemProxySettings();
      NSDictionary *dictProxy = (__bridge_transfer id)proxySettings;
      //是否开启了http代理
      if ([[dictProxy objectForKey:@"HTTPEnable"] boolValue]) {
          NSString *proxyPort = [NSString stringWithFormat: @"%ld", [[dictProxy objectForKey:@"HTTPPort"] integerValue]];
          result(proxyPort);
      }else{
        result(nil);
      }
    }else{
      result(FlutterMethodNotImplemented);
    }
}

@end

File: ios/Classes/HttpProxyPlugin.swift
import Flutter
import UIKit

public class HttpProxyPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "http_proxy", binaryMessenger: registrar.messenger())
    let instance = HttpProxyPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("iOS " + UIDevice.current.systemVersion)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
